lib/game.rb

```diff
@@ -0,0 +1,47 @@
+class Game
+  attr_reader :rolls
+
+  def initialize
+    @rolls = []
+  end
+
+  def roll(pins)
+    @rolls << pins
+  end
+
+  def score
+    total = 0
+    turn = 0
+    10.times do
+      if strike?(turn)
+        total += bonus_score(turn)
+        turn += 1
+      elsif spare?(turn)
+        total += bonus_score(turn)
+        turn += 2
+      else
+        total += frame_score(turn)
+        turn += 2
+      end
+    end
+    total
+  end
+
+  private
```

Comment:
Good practice of keeping some methods private so they won't be called by the user when playing the game.